tests/unit/iontests.js

```diff
@@ -273,30 +273,46 @@ define([
             return fs.readFileSync(path, options);
         }
 
-        function roundTripEventStreams(reader) {
+        function roundTripEventStreams(input) {
             let streams = [];
-            streams.push(new es.IonEventStream(reader));
+            streams.push(new es.IonEventStream(new ion.makeReader(input)));
+            let textWriter = new ion.makeTextWriter();
+            textWriter.writeValues(new ion.makeReader(input));
+            textWriter.close();
+            let text = textWriter.getBytes();
+            streams.push(new es.IonEventStream(new ion.makeReader(text)));
+            let binaryWriter = new ion.makeBinaryWriter();
+            binaryWriter.writeValues(new ion.makeReader(input));
+            binaryWriter.close();
+            let binary = binaryWriter.getBytes();
+            streams.push(new es.IonEventStream(new ion.makeReader(binary)));
```

Comment:
Minor: Some comments in this wall of code would be nice, what is the intention here, we have a stream of the original events, followed by the events from a serialize/read as text, then etc...  Being clear about the combinatorics would be rather nice to have for the next person.

Reply:
i hope the limited comments I added made it a bit more clear.